### ISSUE TITLE
Inconsistent casing in the `dotnet new` template options

### ISSUE THREAD

**Issue by MichaelSimons** [dotnet/cli#5571]
## Steps to reproduce
1.  Install CLI 1.0.0-rc4-004756 on a linux distro (Debian8.2 in my case)
2.  run `dotnet new console --framework netcoreapp1.1`
3.  run `dotnet new mvc --auth None --framework netcoreapp1.1`

## Expected  behavior
I would expect both command to pass.  I would expect all argument names would be lower cased.

## Actual behavior
Step 3 passes but step 2 fails with the following error.
```
Invalid input switch:
  --framework
```
The `console` template will only work with the `--Framework` parameter.

The template help illustrates the inconsistency (see examples at the end)
```
root@56dacbd3ba37:/app7# dotnet new -all
Template Instantiation Commands for .NET Core CLI.

Usage: dotnet new [arguments] [options]

Arguments:
  template  The template to instantiate.

Options:
  -l|--list         List templates containing the specified name.
  -lang|--language  Specifies the language of the template to create
  -n|--name         The name for the output being created. If no name is specified, the name of the current directory is used.
  -o|--output       Location to place the generated output.
  -h|--help         Displays help for this command.
  -all|--show-all   Shows all templates


Templates                                 Short Name       Language      Tags
---------------------------------------------------------------------------------------
Console Application                       console          [C#], F#      Common/Console
Class library                             classlib         [C#], F#      Common/Library
Unit Test Project                         mstest           [C#], F#      Test/MSTest
xUnit Test Project                        xunit            [C#], F#      Test/xUnit
Empty ASP.NET Core Web Application        web              [C#]          Web/Empty
MVC ASP.NET Core Web Application          mvc              [C#], F#      Web/MVC
Web API ASP.NET Core Web Application      webapi           [C#]          Web/WebAPI
Nuget Config                              nugetconfig                    Config
Web Config                                webconfig                      Config
Solution File                             sln                            Solution

Examples:
    dotnet new mvc --auth None --framework netcoreapp1.0
    dotnet new mstest --Framework netcoreapp1.0
    dotnet new --help
```

## Environment data
`dotnet --info` output:

```
.NET Command Line Tools (1.0.0-rc4-004756)

Product Information:
 Version:            1.0.0-rc4-004756
 Commit SHA-1 hash:  ce59e6b19f

Runtime Environment:
 OS Name:     debian
 OS Version:  8
 OS Platform: Linux
 RID:         debian.8-x64
 Base Path:   /usr/share/dotnet/sdk/1.0.0-rc4-004756
```

**Comment by livarcocc** [comment]
cc @mlorbetske. Can you move this to dotnet/templating?

**Comment by mlorbetske** [comment]
This has been fixed with #5560
